taskt/UI/Forms/frmScriptBuilder.cs

```diff
@@ -1815,413 +1405,575 @@ private void lnkGitWiki_LinkClicked(object sender, LinkLabelLinkClickedEventArgs
             //executionManager.ScriptName = new System.IO.FileInfo(ScriptFilePath).Name;
 
-            newEngine.callBackForm = this;
-            newEngine.Show();
-        }
-
-        private void toolStripMenuItem1_Click(object sender, EventArgs e)
+            newEngine.CallBackForm = this;
+            newEngine.Show();
+        }
+        #endregion
+
+        #region Save And Run Tool Strip and Buttons
+        private void saveAndRunToolStripMenuItem_Click(object sender, EventArgs e)
         {
             saveToolStripMenuItem_Click(null, null);
             runToolStripMenuItem_Click(null, null);
-        }
-
-        private void restartApplicationToolStripMenuItem_Click(object sender, EventArgs e)
-        {
-            Application.Restart();
-        }
+        }
+        #endregion
 
-        private void closeApplicationToolStripMenuItem_Click(object sender, EventArgs e)
+        #region Save And Close Buttons
+        private void uiBtnKeep_Click(object sender, EventArgs e)
         {
-            Application.Exit();
+            DialogResult = DialogResult.OK;
         }
 
-        private void showSearchBarToolStripMenuItem_Click(object sender, EventArgs e)
+        private void uiBtnClose_Click(object sender, EventArgs e)
         {
-            //set to empty
-            tsSearchResult.Text = "";
-            tsSearchBox.Text = "";
-
-            //show or hide
-            tsSearchBox.Visible = !tsSearchBox.Visible;
-            tsSearchButton.Visible = !tsSearchButton.Visible;
-            tsSearchResult.Visible = !tsSearchResult.Visible;
-
-            //update verbiage
-            if (tsSearchBox.Visible)
-            {
-                showSearchBarToolStripMenuItem.Text = "Hide Search Bar";
-            }
-            else
-            {
-                showSearchBarToolStripMenuItem.Text = "Show Search Bar";
-            }
-
-        }
-
-        private void frmScriptBuilder_FormClosing(object sender, FormClosingEventArgs e)
-        {
-            if (notifyTray != null)
-            {
-                notifyTray.Visible = false;
-                notifyTray.Dispose();
-            }
-                              
+            DialogResult = DialogResult.Cancel;
         }
-
+        #endregion
+
+        #region View Code Tool Strip
         private void viewCodeToolStripMenuItem_Click(object sender, EventArgs e)
         {
-         
             var currentCommand = lstScriptActions.SelectedItems[0].Tag;
-
-            var jsonText = Newtonsoft.Json.JsonConvert.SerializeObject(currentCommand, new Newtonsoft.Json.JsonSerializerSettings() { TypeNameHandling = Newtonsoft.Json.TypeNameHandling.All });
-
-            var dialog = new Supplemental.frmDialog(jsonText, "Command Code", Supplemental.frmDialog.DialogType.OkOnly, 0);
+            var jsonText = Newtonsoft.Json.JsonConvert.SerializeObject(currentCommand, new Newtonsoft.Json.JsonSerializerSettings()

```

Comment:
Can you please shorten this namespace `Newtonsoft.Json` reference

Reply:
done